### ISSUE TITLE
must be a problem with walmart link/tracking

### ISSUE THREAD

**Issue by robgpeak** [robgpeak/CompLinks-xclub#334]
@cgladue have a look at all the walmart transactions, and you can see that the majority of them are NOT being tracked properly and are ending up as orphan transaction.

https://xclub.rewardseverywhere.co/dashboard/admin/profitability

**Comment by robgpeak** [comment]
This one below was mine.  Here was my order number 5991981-882937

<img width="1035" alt="Screen Shot 2019-11-27 at 8 59 08 AM" src="https://user-images.githubusercontent.com/28393235/69729839-ffece000-10f4-11ea-9847-81ae1c5e1b40.png">

**Comment by cgladue** [comment]
we may need to escalate to Impact, because i dont see anything wrong, maybe walmart is having issues or theres something that changed for them that didnt for others ?  because there are some walmart ones that are OK and there are plenty of other stores (Home depot, Kohls, Target) that are fine and the interstitial dosnt care about what store your shopping it does the same thing every time.

**Comment by cgladue** [comment]
in the case of your trip, did you do anything different then you usually do ?

**Comment by robgpeak** [comment]
no, nothing different on my end.

**Comment by cgladue** [comment]
ok k can we open a ticket with impact then to take a look, i went through everything that i can on my side and it all seems fine.  Walmart just switched to Impact on October 7th so it was recent but not recent enough to think that had anything to do with it

**Comment by cgladue** [comment]
were you using the extension? 

from the website and on desktop or mobile ?

**Comment by robgpeak** [comment]
did not use the extension and it was from my laptop

>

**Comment by robgpeak** [comment]
Let me start another trip and buy something and we'll track it together
through a screencast...

>

**Comment by cgladue** [comment]
i dont know the right place to open an issue for the extension, and its not related to this, but when i click a link for the search results to start a trip it looks like it goes directly to the advertiser not going through us first so we would never get a comission ... see if i am right ?

**Comment by robgpeak** [comment]
I will check now, but in the mean time, have a look at this
https://www.loom.com/share/c736717e0f964d1296fc26d5c2e5422d

Here is the *Order #:* 6021995-333140

**Comment by robgpeak** [comment]
The link in the search result worked as expected for me.  Have a look https://www.loom.com/share/3cd7e48cfdd34513887991857a2ab39a

**Comment by cgladue** [comment]
ooo ok i thought the link would start the trip automatically for you not click twice, ok nevermind

**Comment by cgladue** [comment]
ok i dont see the purchase in the Comission API yet ill keep an eye on it

**Comment by robgpeak** [comment]
unfortunately it might not show up until the item is picked up. (Dec. 10)
but do you get a record of an order being made in the API?

**Comment by cgladue** [comment]
no, it may not untill you pick it up

**Comment by cgladue** [comment]
ill keep checking every so often tho and let you know when it shows up .. the process looks correct tho without anything obvious

**Comment by robgpeak** [comment]
This one was also mine...

11/22/2019 02/01/2020 100105
<https://xclub.rewardseverywhere.co/dashboard/users/view/100105?tab=tab-purchases>
CompLinks 9383.4342.464888 Walmart.com 9.99 0.10 0.05

>

**Comment by robgpeak** [comment]
I went into the back end of impact and looked at the reports for walmart and found this.

https://www.loom.com/share/1b8e1dd0950d4557b90c2556a71cec4b

Here is the link in the backend of impact that is shows was clicked through to walmart.
https://xclub.rewardseverywhere.co/trip/startNew?aid=13861&lid=0&url=%2f%2fgoto.walmart.com%2fc%2f406688%2f612734%2f9383%3fsourceid%3dimp_000011112222333344%26veh%3daff

**Comment by cgladue** [comment]
```
 <Action>
      <Id>9383.4348.1386513</Id>
      <CampaignId>9383</CampaignId>
      <CampaignName>Walmart Affiliate Program</CampaignName>
      <ActionTrackerId>16681</ActionTrackerId>
      <ActionTrackerName>Online Sale</ActionTrackerName>
      <State>PENDING</State>
      <AdId>612734</AdId>
      <Payout>0.32</Payout>
      <DeltaPayout>0.32</DeltaPayout>
      <IntendedPayout>0.32</IntendedPayout>
      <Amount>7.88</Amount>
      <DeltaAmount>7.88</DeltaAmount>
      <IntendedAmount>7.88</IntendedAmount>
      <Currency>USD</Currency>
      <ReferringDate>2019-11-27T10:11:10-05:00</ReferringDate>
      <EventDate>2019-11-27T10:13:54-05:00</EventDate>
      <CreationDate>2019-11-28T20:03:20-05:00</CreationDate>
      <LockingDate>2020-02-01T03:00:00-05:00</LockingDate>
      <ClearedDate/>
      <ReferringType>CLICK_COOKIE</ReferringType>
      <ReferringDomain>xclub.rewardseverywhere.co</ReferringDomain>
      <PromoCode/>
      <Oid>6021995333140</Oid>
      <CustomerArea>n/a</CustomerArea>
      <CustomerCity>Wellington</CustomerCity>
      <CustomerRegion>Florida</CustomerRegion>
      <CustomerCountry>US</CustomerCountry>
      <SubId1/>
      <SubId2/>
      <SubId3/>
      <SharedId/>
      <Uri>/Mediapartners/IRpheWJsW4ov406688Y2yse2g6i5P5gdh1/Actions/9383.4348.1386513</Uri>
    </Action>
```

Your order finally appeared, but no trip ID was reported ... how do we escalate this to impact ?

**Comment by robgpeak** [comment]
I will submit a ticket with them.  So basically I will copy and paste what
you sent me above and tell them

"There is missing data in the API feed.

Here is what we received, (paste what you sent) but we are missing the trip
ID field.  Please look I to this and let us know when it can be fixed."

Does that look about right?

**Comment by cgladue** [comment]
You can add that we use subid1 field that should be populated like the
other stores do. And that other advertisers are working ok. Just this one
thats causing trouble.

**Comment by robgpeak** [comment]
So the sub ID field is the trip ID yes?

**Comment by cgladue** [comment]
Yea they have three fields we can use subid1,2,3. We use subid1 to pass our
tripid

**Comment by robgpeak** [comment]
👍

**Comment by robgpeak** [comment]
Were you testing a bunch of links to walmart with subid's that were 3000, 3001, 3002, 4000 etc.

I did those 4 test yesterday, but those were not the shopping IDs I noted...

Very strange.  You can also see that just yesterday it has "8" clicks registered with NO subid1 values, and I am guessing at least 4 of those were mine.

<img width="1194" alt="Screen Shot 2019-12-04 at 8 18 20 AM" src="https://user-images.githubusercontent.com/28393235/70145773-d8ef5a80-166e-11ea-8c9c-26e977454e21.png">

**Comment by cgladue** [comment]
Yes it was me.  Trying to see a pattern

Ill email you when i get into work. But all the links i tested that i
directly typed jnto the browser worked and the ones that i let the site
redirect me to didnt.  I think we may need to require that the user click
to continue for some stores.

**Comment by robgpeak** [comment]
but why (seemingly) all of a sudden, and also why just walmart.  other ones seem to be working fine, and prior to mid month-ish in November, everything was working fine.

**Comment by cgladue** [comment]
Maybe because they are with impact now

**Comment by robgpeak** [comment]
Again, why would it only be walmart on impact that is doing this?  and... "I think we may need to require that the user click to continue for some stores..." that would suck... and when you say some stores, does that mean it is happening for other stores that you know of?

**Comment by cgladue** [comment]
its not all walmart either, its some walmart and no, i dont see any others currently but thought it should be an option .. i have more testing to do tonight to try to find a pattern to know for sure.

**Comment by robgpeak** [comment]
I know you are still working on a possible solution, but is it possible to revers engineer what impact/walmart is saying was the referring URL for us to see on our side who started the shopping trip?

https://xclub.rewardseverywhere.co/trip/startNew?aid=13861&lid=0&url=%2f%2fgoto.walmart.com%2fc%2f406688%2f612734%2f9383%3fsourceid%3dimp_000011112222333344%26veh%3daff

**Comment by robgpeak** [comment]
<img width="998" alt="Screen Shot 2019-12-06 at 12 30 52 PM" src="https://user-images.githubusercontent.com/28393235/70343042-44c6f400-1824-11ea-888e-e2045b75ebcb.png">

**Comment by cgladue** [comment]
this is barbara manzo i believe she had a trip to walmart at the same time roughly as the reporting says that the referral took place:

![image](https://user-images.githubusercontent.com/7914133/70375675-9035dc80-18ce-11ea-96ac-1b50692f69e3.png)

**Comment by robgpeak** [comment]
Ok please give her the points.

Thanks,

Rob

**Sent from my phone

**Comment by cgladue** [comment]
all set:

![image](https://user-images.githubusercontent.com/7914133/70376006-9037db80-18d2-11ea-9bfa-39ac721c3b31.png)

**Comment by cgladue** [comment]
Rob i tested every concievable combination of links and formats and the only thing i can see is that it only reports correctly when i type ir (or paste it) directly into the browser, if i let it redirect me then it fails to report correctly....  not sure why this is happening only to walmart

**Comment by robgpeak** [comment]
I don't understand how that is even possible.  When it redirects, what is
the URL that is ultimately sent through to Walmart?  Meaning, isn't it the
same exact address that it would be if you manually typed it in?

Thanks,

Rob

**Sent from my phone

**Comment by cgladue** [comment]
yes its the same link... but i tested it a dozen times and thats all i can see as the common factor

**Comment by robgpeak** [comment]
And yet other links from our site to Impact's tracking for other merchants
is doing the same sort of redirect and they ARE tracking properly, yeah?

Thanks,

Rob

**Sent from my phone

**Comment by cgladue** [comment]
Yea, in the last few days there has been Petco, Kohls and Home Depot all tracked properly and thats the same page that does the redirects for all the networks .. theres only 1 page that handles all the redirects.

![image](https://user-images.githubusercontent.com/7914133/70377405-20315180-18e2-11ea-9c0e-239fb634f376.png)

**Comment by cgladue** [comment]
thats why it seems to me for all i can tell to be an issue on Walmarts side.. but you would think there would be a lot more people complaining ...

**Comment by robgpeak** [comment]
Can you properly articulate the problem so when I send it to their customer
service they would be able to make better sense of it other than just
telling us to tested again. Meaning we need to be able to show them exactly
what the issue is and how it compares to other tracking for other impact
radius sites that are working properly based on the redirection.

Thanks,

Rob

**Sent from my phone

**Comment by cgladue** [comment]
yea here you go:


From our testing only the links that are DIRECT visits (i.e. no auto-redirect, they have to be a physical link click or typed or pasted directly into the browser URL field) are working, see below for my tests:


subId1 **did not** show up in Impact.com dashboard for these tests, these tests all used javascript to redirect the user after 5 seconds (the way its always been):

Javscript Test
window.location.href = ('https://goto.walmart.com/c/406688/612734/9383?sourceid=imp_000011112222333344&veh=aff&subId1=4005');

Incognito, Main Page Shop Now Button:
https://goto.walmart.com/c/406688/612734/9383?sourceid=imp_000011112222333344&veh=aff&subId1=7877

Incognito, Store Page, Free 2 Day Shipping Link
https://goto.walmart.com/c/406688/619711/9383?sourceid=imp_000011112222333344&veh=aff&subId1=7878



These tests **DID** pass the subId1 field through to impact.com dashboard as it should, they were either directly pasted into the URL bar or the text link on the redirect page was clicked:

Incognito, direct Parameter at end
https://goto.walmart.com/c/406688/612734/9383?sourceid=imp_000011112222333344&veh=aff&subId1=4002

Incognito, direct Parameter at start
https://goto.walmart.com/c/406688/612734/9383?subId1=4003&sourceid=imp_000011112222333344&veh=aff

Incognito, Main Page Shop Now Button, SubId at beginning, direct:
https://goto.walmart.com/c/406688/612734/9383?subId1=4000&sourceid=imp_000011112222333344&veh=aff

Incognito, Store Page, Free 2 Day Shipping Link, direct
https://goto.walmart.com/c/406688/619711/9383?subId1=4001&sourceid=imp_000011112222333344&veh=aff

direct, didnt work before:
https://goto.walmart.com/c/406688/612734/9383?sourceid=imp_000011112222333344&veh=aff&subId1=4004


From my testing all the ones that were the result of a manual click on a HTML link or typed into the browser directly worked as expected, and the ones that relied on javascript to automatically redirect the user to the url all failed.

**Comment by robgpeak** [comment]
Did you post this walmart purchases to this guys account, or did that work automatically?

One other thing to note that is different however, is the fact that there is no purchase data like there is normally...
![Screen Shot 2019-12-11 at 3 54 41 PM](https://user-images.githubusercontent.com/28393235/70659669-927d9b00-1c2e-11ea-99bc-2f47b05c67b1.png)

**Comment by cgladue** [comment]
No i didnt do anything ... how do you know those trips werent assigned to him always ?

**Comment by cgladue** [comment]
and unfortunatly i think that the change in product reporting is due to the fact they are part of Impact now and impact isnt reporting it

**Comment by robgpeak** [comment]
I didn't know they weren't attached to him but I'm just saying that based
on the recent dates of the transactions it seems that they're showing up
with the proper shopping trip ID and he's getting. Meaning they're not
orphan transaction.

As for not showing product reporting, that fucking blows!  Product
reporting is a key feature in the overall value proposition in selling the
CompLinks Rewards Everywhere platform to potential clients...  Can you dig
around and see if there is such a thing in Impact?

>

**Comment by cgladue** [comment]
In the action report i can see the products i think.  But its a manual
report not avial le in the APi.  But its been the case since day #1 that
they don't make that data available via the APi and ive been back and forth
with them over it before.  I can open another case with rhem if you would
like to see if they have made anything new available.

**Comment by robgpeak** [comment]
please do.

>

**Comment by cgladue** [comment]
https://help.impactradius.com/hc/en-us/requests/149612 

opened

**Comment by cgladue** [comment]
still nothing came of this they dont seem to care ...

**Comment by robgpeak** [comment]
let me see if I can follow up with them again

>

**Comment by robgpeak** [comment]
I just placed an order from the shop now button on the home page of xclub.rewardseverywhere.co Shopping trip ID 8000) and here is the order number... 6251935780559 Supposed to be delivered tomorrow.

**Comment by robgpeak** [comment]
A bit of good news, Walmart seems to have tracked a trip that I made... (Trip ID 8000) and the points were credited to my account automatically.  The bad news is there is still no product description.  Still waiting on a reply from Impact again.

**Comment by cgladue** [comment]
they asked for an actual link we are using yesterday, i gave them one and they are still investigating aparently.

**Comment by cgladue** [comment]
Rob, according to them this seems to have been corrected somehow since december 11th do you share that feeling ?  have you seen anything else show up thats unassigned ?

**Comment by robgpeak** [comment]
There does not appear to be any orphan transactions.  How about the
purchase descriptions...(the names of the items someone buys)?  Is that a
separate issue?

Thanks,

Rob

**Sent from my phone

**Comment by cgladue** [comment]
yea thats separate,  and i dont think they ever replied to you on that part
of the question now i think of it

**Comment by robgpeak** [comment]
Please ask them again in a separate thread.

Thanks,

Rob

**Sent from my phone

**Comment by cgladue** [comment]
i followed up again with a request for update from these regarding this .. waiting for a response

**Comment by robgpeak** [comment]
Ok

Thanks,

Rob

**Sent from my phone
